pkg/controller/framework/informers/factory.go

```diff
@@ -27,6 +28,56 @@ import (
 	"k8s.io/kubernetes/pkg/watch"
 )
 
+type SharedInformerFactory interface {
+	Start(time.Duration)
+
+	Pod() framework.SharedIndexInformer
+	Node() framework.SharedIndexInformer
+	Namespace() framework.SharedIndexInformer
+	PersistentVolumeClaim() framework.SharedIndexInformer
+	PersistentVolume() framework.SharedIndexInformer
+}
+
+type sharedInformerFactory struct {
+	client    clientset.Interface
+	informers map[reflect.Type]framework.SharedIndexInformer
+}
+
+func NewSharedInformerFactory(client clientset.Interface) SharedInformerFactory {
+	return &sharedInformerFactory{
+		client:    client,
+		informers: make(map[reflect.Type]framework.SharedIndexInformer),
+	}
+}
+
+func (s *sharedInformerFactory) Start(resyncPeriod time.Duration) {
```

Comment:
This should be re-entrant to handle cases where it gets called more than once.  That can happen in bootstrapping cases and in cases where we have late joiners.